File: frontend/src/types/career.ts
export type CareerPlan = {
  id: number;
  name: string;
  current_role: string;
  target_role: string;
  skills: string[];
  readiness_score: number;
  roadmap: string[];
};

export type ResumeResult = {
  id: number;
  score: number;
  strengths: string[];
  gaps: string[];
  suggestions: string[];
};

export type JobMatch = {
  title: string;
  match_score: number;
  reason: string;
  missing_skills: string[];
};

export type DomainScore = {
  domain: string;
  score: number;
  confidence: string;
  evidence: string[];
  missing_keywords: string[];
};

export type ResumeIntelligence = {
  id: number;
  file_name: string;
  target_role: string;
  overall_score: number;
  ats_score: number;
  seniority_signal: string;
  detected_skills: string[];
  missing_skills: string[];
  nlp_keywords: string[];
  domain_ranking: DomainScore[];
  deep_learning_signal: number;
  job_matches: JobMatch[];
  resume_strengths: string[];
  priority_fixes: string[];
  rewritten_bullets: string[];
  project_recommendations: string[];
  interview_focus: string[];
  learning_plan: string[];
};

export type ChatResult = {
  answer: string;
  recommended_actions: string[];
};

export type InterviewResult = {
  id: number;
  score: number;
  feedback: string;
};

export type InterviewBatch = {
  domain: string;
  difficulty: string;
  questions: string[];
};

export type User = {
  id: number;
  name: string;
  email: string;
  phone: string | null;
  is_email_verified: boolean;
  is_phone_verified: boolean;
};

export type AuthResponse = {
  access_token: string;
  token_type: string;
  user: User;
  dev_otp?: string | null;
};

export type EmotionResult = {
  emotion: string;
  sentiment: string;
  tone: string;
  confidence: number;
  scores: Record<string, number>;
  source: string;
  recommendation: string;
};

export type PersonalityReport = {
  personality_type: string;
  communication_style: string;
  learning_style: string;
  career_matches: string[];
  best_work_environment: string;
  strengths: string[];
  weaknesses: string[];
  scores: Record<string, number>;
  report: string;
};

File: backend/app/schemas/career.py
from pydantic import BaseModel, Field


class CareerPlanCreate(BaseModel):
    name: str = Field(min_length=2, max_length=120)
    current_role: str = Field(min_length=2, max_length=160)
    target_role: str = Field(min_length=2, max_length=160)
    skills: list[str] = Field(default_factory=list)
    experience_years: float = Field(ge=0, le=50)
    projects_count: int = Field(ge=0, le=100)
    interview_confidence: int = Field(ge=1, le=10)


class CareerPlanOut(BaseModel):
    id: int
    name: str
    current_role: str
    target_role: str
    skills: list[str]
    readiness_score: float
    roadmap: list[str]


class ChatMessage(BaseModel):
    message: str = Field(min_length=1)
    target_role: str | None = None


class ChatResponse(BaseModel):
    answer: str
    recommended_actions: list[str]


class ResumeAnalyzeRequest(BaseModel):
    resume_text: str = Field(min_length=30)
    target_role: str = Field(min_length=2, max_length=160)


class ResumeAnalyzeResponse(BaseModel):
    id: int
    score: float
    strengths: list[str]
    gaps: list[str]
    suggestions: list[str]


class JobMatch(BaseModel):
    title: str
    match_score: float
    reason: str
    missing_skills: list[str]


class DomainScore(BaseModel):
    domain: str
    score: float
    confidence: str
    evidence: list[str]
    missing_keywords: list[str]


class ResumeIntelligenceResponse(BaseModel):
    id: int
    file_name: str
    target_role: str
    overall_score: float
    ats_score: float
    seniority_signal: str
    detected_skills: list[str]
    missing_skills: list[str]
    nlp_keywords: list[str]
    domain_ranking: list[DomainScore]
    deep_learning_signal: float
    job_matches: list[JobMatch]
    resume_strengths: list[str]
    priority_fixes: list[str]
    rewritten_bullets: list[str]
    project_recommendations: list[str]
    interview_focus: list[str]
    learning_plan: list[str]


class InterviewQuestionRequest(BaseModel):
    target_role: str = Field(min_length=2, max_length=160)
    difficulty: str = "mid"


class InterviewQuestionResponse(BaseModel):
    question: str


class InterviewBatchRequest(BaseModel):
    domain: str = "AI / Machine Learning"
    difficulty: str = "mid"
    count: int = Field(default=7, ge=5, le=10)


class InterviewBatchResponse(BaseModel):
    domain: str
    difficulty: str
    questions: list[str]


class InterviewEvaluateRequest(BaseModel):
    target_role: str = Field(min_length=2, max_length=160)
    question: str = Field(min_length=5)
    answer: str = Field(min_length=5)


class InterviewEvaluateResponse(BaseModel):
    id: int
    score: float
    feedback: str


class ReadinessPredictRequest(BaseModel):
    experience_years: float = Field(ge=0, le=50)
    projects_count: int = Field(ge=0, le=100)
    skill_match_percent: float = Field(ge=0, le=100)
    interview_confidence: int = Field(ge=1, le=10)


class ReadinessPredictResponse(BaseModel):
    readiness_score: float
    label: str
